spot-ml/src/main/scala/org/apache/spot/SpotLDAWrapper.scala

```diff
@@ -0,0 +1,89 @@
+package org.apache.spot
+
+import java.io.{File, PrintWriter}
+
+import org.apache.spark.rdd.RDD
+import org.apache.spot.{SpotLDACWrapper, SpotSparkLDAWrapper}
+import scala.io.Source._
+import scala.sys.process._
+import org.apache.log4j.Logger
+
+/**
+  * Contains routines for LDA including pre and post operations
+  * 1. Creates list of unique documents, words and model based on those two
+  * 2. Processes the model calling MPI
+  * 3. Reads MPI results: Topic distributions per document and words per topic
+  * 4. Calculates and returns probability of word given topic: p(w|z)
+  */
+
+object SpotLDAWrapper {
+
+
+  case class SpotLDAInput(doc: String, word: String, count: Int) extends Serializable
+
+  case class SpotLDAOutput(docToTopicMix: Map[String, Array[Double]], wordResults: Map[String, Array[Double]])
+
+  case class InvalidLDAImp(message: String) extends Exception(message)
+
+  def runLDA(docWordCount: RDD[SpotLDAInput],
+             modelFile: String,
+             topicDocumentFile: String,
+             topicWordFile: String,
+             mpiPreparationCmd: String,
+             mpiCmd: String,
+             mpiProcessCount: String,
+             topicCount: Int,
+             localPath: String,
+             ldaPath: String,
+             localUser: String,
+             dataSource: String,
+             nodes: String,
+             ldaImp: String,
+             logger: Logger,
+             ldaOptimizer: String,
+             ldaAlpha: Double,
+             ldaBeta: Double,
+             maxIterations: Int,
+             prgSeed: Option[Long]):   SpotLDAOutput =  {
+
+    val spotOutput = ldaImp match {
+      case "LDAC" => SpotLDACWrapper.runLDA(docWordCount,
+        modelFile,
+        topicDocumentFile,
+        topicWordFile,
+        mpiPreparationCmd,
+        mpiCmd,
+        mpiProcessCount,
+        topicCount,
+        localPath,
+        ldaPath,
+        localUser,
+        dataSource,
+        nodes,
+        prgSeed)
+      case "SparkLDA" => SpotSparkLDAWrapper.runLDA(docWordCount,
+        topicCount,
+        logger,
+        prgSeed,
+        ldaOptimizer,
+        ldaAlpha,
+        ldaBeta,
+        maxIterations)
+      case _ => throw InvalidLDAImp(s"Invalid LDA implementation specified, options are LDAC or SparkLDA")
+    }
+
+    spotOutput
+  }
+  /**
```

Comment:
it seems like this comment is a stray... should it be merged with the one up top?